client/my-sites/controller.js

```diff
@@ -336,7 +336,7 @@ export function noSite( context, next ) {
  */
 export function siteSelection( context, next ) {
 	const { getState, dispatch } = getStore( context );
-	const siteFragment = context.params.site || getSiteFragment( context.path );
+	const siteFragment = context.params.site || context.query.site || getSiteFragment( context.path );
```

Comment:
Some redirects use `/pricing?site=:site` as a destination. We need to handle this case as well. Would there be any side effect?